src/Microsoft.Azure.WebJobs/ParameterBindingData.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Text.Json.Serialization;
+
+namespace Microsoft.Azure.WebJobs
+{
+    /// <summary>
+    /// A reference type for supporting SDK-type bindings in out-of-process Function workers
+    /// This type has built-in serialization using System.Text.Json.
+    /// </summary>
+    [JsonConverter(typeof(ParameterBindingData))]
```

Comment:
I might be missing something, but I don't think this is how `JsonConverterAttribute` is used. Additionally, how is this type expected to be serialized? I think it can only be written to JSON, not read from it as is. It will need some work to know which ctor to use for deserialization.

Reply:
I honestly just copied the CloudEvent format and changed it to work with the parameter binding data schema we are designing: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/Messaging/CloudEvent.cs

They also include a [deserialization](https://github.com/Azure/azure-sdk-for-net/blob/6ea1458cc053a2061e8846fedb6d03c0df61212f/sdk/core/Azure.Core/src/Messaging/CloudEvent.cs#L198) method  and we could do something similar?

Reply:
Also the value we are passing in the constructor is incorrect. A valid value should be the type of an actual converter implementation (A class which derives from [JsonConverter ](https://docs.microsoft.com/en-us/dotnet/api/system.text.json.serialization.jsonconverter-1?view=net-6.0)and override the `Read` and `Write` methods as needed). 

I think this attribute decoration can be removed. If we want to **_customize_** the serialization/desreialization, we will have to write a converter implementation and use that.

Reply:
Yeah if you want to support serialization and deserialization, this type looks simple enough to do it without needing a custom converter. I think what you will need is:

1. Add `[JsonConstructor]` to this constructor: `public ParameterBindingData(string source, BinaryData content, string dataContentType)`
2. Update the above constructor's parameters to match the properties they map to: `public ParameterBindingData(string source, BinaryData content, string contentType)`